examples/solvers/admm_spectral_tomography.py

```diff
@@ -0,0 +1,169 @@
+"""Total variation spectral tomography using preconditioned nonlinear ADMM.
+
+In this example we solve the optimization problem
+
+    min_x  ||A(x) - y||_2^2 + lam * ||grad(x)||_{nuc, 1, 2}
+
+where ``||.||_{nuc, 1, 2}`` is the nuclear L1-L2 norm and ``A`` a 2 x 4
+operator matrix
+
+        (c_1 * exp(-R)  c_2 * exp(-R)                              )
+    A = (                              c_3 * exp(-R)  c_4 * exp(-R))
+
+with constants ``c_i`` and the parallel beam ray transform ``R``. In other
+words, ``A`` acts on an input ``x`` with 4 components as follows:
+
+           (c_1 * exp(-R(x_1)) + c_2 * exp(-R(x_2)))
+    A(x) = (c_3 * exp(-R(x_3)) + c_4 * exp(-R(x_4)))
+
+This reflects a simplified model for 2D spectral CT with 4 (discrete) beam
+energies and 2 detector energy bins.
+
+The problem is rewritten in decoupled form as
+
+    min_x g(L(x))
+
+with a separable sum ``g`` of functionals and the stacked operator ``L``:
+
+    g(z) = ||z_1 - g||_2^2 + lam * ||z_2||_1,
+
+               ( A(x)    )
+    z = L(x) = ( grad(x) ).
+
+See the documentation of the `admm_precon_nonlinear` solver for further
+details.
+Note that this problem is underdetermined as a simple count of input
+channels vs. data channels shows. This can be changed in the code below
+(along with ``souce_spectrum`` and ``energy_factors``).
+"""
+
+import numpy as np
+import odl
+
+# --- Set up the forward operator --- #
+
+# Space of a monochromatic phantom (volume)
+single_energy_space = odl.uniform_discr(
+    min_pt=[-20, -20], max_pt=[20, 20], shape=[100, 100], dtype='float32')
+reco_space = single_energy_space ** 2
+
+# Create a parallel beam geometry with flat detector, using 180 angles
+geometry = odl.tomo.parallel_beam_geometry(single_energy_space, num_angles=180)
+
+# Define number of energy bins on both sides as well as the source spectrum
+num_bins_reco = 4
+num_bins_data = 2
+assert num_bins_reco >= num_bins_data
+assert num_bins_reco % num_bins_data == 0
+group_size = num_bins_reco // num_bins_data
+# Note: the spectrum (i.e. intensity per channel) is additive on the data
+# side, i.e., adding more entries adds to the total signal in the data bins
+source_spectrum = [8e4, 1e5, 8e4, 6e4]
```

Comment:
why soo high values?

Reply:
I'll have to work a bit on this example anyway. Currently it's a bit silly since for `n x n` problems, you can actually linearize.